docgenerator/src/main/scala/codacy/codenarc/docs/DocGenerator.scala

```diff
@@ -0,0 +1,311 @@
+
+  private val defaultPriority = "3"
+  private val defaultRuleType = "CodeStyle"
+  private val defaultRulePriority = RulePriority(Some(defaultPriority), Some(defaultRuleType))
+
+  /**
+    * Converts from CodeNarc's complexity to Codacy level
+    */
+  val levelFromPriority: PartialFunction[Option[String], Result.Level.Value] = {
+    case Some("1") => Result.Level.Err
+    case Some("2") => Result.Level.Warn
+    case Some("3") => Result.Level.Info
+    case _ => Result.Level.Info
+  }
+
+  val categoryTypeFromCategory: PartialFunction[Option[String], Pattern.Category.Value] = {
+    case Some("security") => Pattern.Category.Security
+    case Some("unused") => Pattern.Category.UnusedCode
+    case _ => Pattern.Category.CodeStyle
+  }
+
+  /**
+    * Convert from list of parameters to Parameter.Specification list
+    */
+  def parameterSpecificationFromParametersStringList(
+      parameters: Option[Set[String]]
+  ): Option[Set[Parameter.Specification]] =
+    parameters.map(
+      paramList => paramList.map(param => Parameter.Specification(Parameter.Name(param), Parameter.Value("")))
+    )
+
+  def parameterDescriptionFromParametersStringList(
+      parameters: Option[Set[String]]
+  ): Option[Set[Parameter.Description]] =
+    parameters.map(
+      paramList => paramList.map(param => Parameter.Description(Parameter.Name(param), Parameter.DescriptionText("")))
+    )
+
+  /**
+    * returns the regex to get information from markdown file for a rule
+    *
+    * @param ruleName Name of the rule to get information
+    * @return
+    */
+  def ruleInfoRegex(ruleName: String) = new Regex("## " + ruleName + "([\\n\\S\\s\\w\\d][^##])*")
+
+  /**
+    * Gets the tool version from the pattern.json file
+    *
+    * @param args
+    * @return
+    */
+  def versionFromArgs(args: Array[String]): String =
+    args.headOption
+      .orElse {
+        val sourceFile = io.Source.fromFile(s"$codeNarcVersionFile")
+        val resourceContent = sourceFile.getLines.mkString("")
+        sourceFile.close()
+        Some(resourceContent)
+      }
+      .getOrElse {
+        throw new Exception("No version provided")
+      }
+
+  /**
+    * Gets the documentation markdown name from the href element value
+    *
+    * @param href
+    * @return
+    */
+  def extractMdFilenameFromHref(href: String): String =
+    href.replaceAllLiterally("./", "").replaceAllLiterally(".html", "").split("#")(0) + ".md"
+
+  /**
+    * Get information about a rule from markdown file
+    *
+    * @param docsFolder   Documentation folder
+    * @param ruleName     Name of the rule to get information about
+    * @param ruleFilename The name of the markdown file where rule information is stored at.
+    * @return
+    */
+  def getRuleExtendedInfoFromMarkdown(docsFolder: String, ruleName: String, ruleFilename: String): String = {
+    val ruleInfoMarkdown = File(s"$docsFolder$ruleFilename").contentAsString
```

Comment:
Better use better files to interpolate paths here:
https://github.com/pathikrit/better-files#instantiation
Something like:
```scala
val ruleInfoMarkdown = (root / docsFolder / ruleFilename).contentAsString 
```

Reply:
Done ✅